pkg/agents/team5/social.go

```diff
@@ -77,11 +78,20 @@ func (a *CustomAgent5) calculateAverageFavour() int {
 }
 
 func (a *CustomAgent5) incrementDaysSinceLastSeen() {
-	for _, mem := range a.socialMemory {
+	for id := range a.socialMemory {
+		mem := a.socialMemory[id]
 		mem.daysSinceLastSeen++
+		a.socialMemory[id] = mem
 	}
 }
 
+// TODO: Consider adding check for exists and if not exists then add to memory.
+func (a *CustomAgent5) resetDaysSinceLastSeen(id uuid.UUID) {
+	mem := a.socialMemory[id]
+	mem.daysSinceLastSeen = 0
+	a.socialMemory[id] = mem
+}
+
 // TODO: Consider adding check for exists and if not exists then add to memory.
 func (a *CustomAgent5) resetSocialKnowledge(id uuid.UUID) {
 	mem := a.socialMemory[id]
```

Comment:
What's the behavior if not existing in memory?

Reply:
Please see 'Added safety checks for memory existing in social functions' commit for fix